switchapi/src/switch_packet.c

```diff
@@ -410,3 +597,415 @@ int start_switch_api_packet_driver()
+    if (handle_type == SWITCH_HANDLE_TYPE_LAG) {
+        lag_info = switch_api_lag_get_internal(rx_key->port_lag_handle);
+        if (!lag_info) {
+            SWITCH_API_ERROR("invalid lag handle");
+            return SWITCH_STATUS_INVALID_HANDLE;
+        }
+        ifindex = lag_info->ifindex;
+    } else {
+        port_info = switch_api_port_get_internal(rx_key->port_lag_handle);
+        if (!port_info) {
+            SWITCH_API_ERROR("invalid port handle");
+            return SWITCH_STATUS_INVALID_HANDLE;
+        }
+        ifindex = port_info->ifindex;
+    }
+
+    bd_handle = rx_key->handle;
+    handle_type = switch_handle_get_type(rx_key->handle);
+    if (handle_type == SWITCH_HANDLE_TYPE_INTERFACE) {
+        intf_info = switch_api_interface_get(rx_key->handle);
+        if (!intf_info) {
+            SWITCH_API_ERROR("intf_handle %lx is invalid", rx_key->handle);
+            return SWITCH_STATUS_INVALID_HANDLE;
+        }
+        if (!SWITCH_INTF_IS_PORT_L3(intf_info)) {
+            SWITCH_API_ERROR("intf_handle %lx is not l3", rx_key->handle);
+            return SWITCH_STATUS_INVALID_HANDLE;
+        }
+        bd_handle = intf_info->bd_handle;
+    }
+
+    bd_info = switch_bd_get(bd_handle);
+    if (!bd_info) {
+        SWITCH_API_ERROR("bd derivation failed %lx", rx_key->handle);
+        return SWITCH_STATUS_INVALID_HANDLE;
+    }
+
+    node = tommy_list_head(&packet_rx_list);
+    while (node) {
+        tmp_rx_info = (switch_packet_rx_info_t *) node->data;
+        tmp_rx_entry = &tmp_rx_info->rx_entry;
+        if (tmp_rx_entry->bd == handle_to_id(bd_handle) &&
+            tmp_rx_entry->ifindex == ifindex &&
+            tmp_rx_entry->port == handle_to_id(rx_key->port_handle) &&
+            tmp_rx_entry->reason_code == rx_key->reason_code &&
+            tmp_rx_entry->priority == rx_key->priority) {
+            node_found = TRUE;
+            break;
+        }
+        node = node->next;
+    }
+
+    if (!node_found) {
+        SWITCH_API_ERROR("tx filter delete failed. node find failed");
+        return SWITCH_STATUS_ITEM_NOT_FOUND;
+    }
+
+    tommy_list_remove_existing(&packet_rx_list, node);
+
+    switch_free(rx_info);
+    return status;
+}
+
+switch_status_t
+switch_packet_rx_info_get(
+        switch_packet_rx_entry_t *rx_entry,
+        switch_packet_rx_info_t **rx_info)
+{
+    switch_packet_rx_info_t           *tmp_rx_info = NULL;
+    tommy_node                        *node = NULL;
+    switch_packet_rx_entry_t          *tmp_rx_entry = NULL;
+    switch_status_t                    status = SWITCH_STATUS_ITEM_NOT_FOUND;
+
+    *rx_info = NULL;
+
+    node = tommy_list_head(&packet_rx_list);
+    while (node) {
+        tmp_rx_info = (switch_packet_rx_info_t *) node->data;
+        tmp_rx_entry = &tmp_rx_info->rx_entry;
+        if (((tmp_rx_entry->port_valid && tmp_rx_entry->port == rx_entry->port) ||
+             !tmp_rx_entry->port_valid) && 
```

Comment:
Lines 963 and 964 can be simplyfied as (?) -
(! tmp_rx_entry->port_valid || (tmp_rx_entry->port == rx_entry->port)) && ...
and same for few following lines.